docs/api_compatibility.md

```diff
@@ -0,0 +1,47 @@
+# API compatibility
+
+JAX is constantly evolving, and we are not yet ready to commit to a stable API.
+That said, we want to minimize churn for the JAX user community.
+
+JAX follows a 3 month deprecation policy. When an incompatible change is made
```

Comment:
3 months is fast for an OSS project. The challenge is that you need to wait for downstream libraries to release fixes, and that can take time. It's not just about giving end-users time to upgrade.

6 months would be more in line with general norms. Definitely up to you, though, there are many tradeoffs here.

Reply:
I also think that 3 months is fast. I expect that a lower bound on that interval will be our ability to migrate internal Google users off an API end point. Most likely a widely used API point will take much longer to remove.

Reply:
I left the deprecation period at three months with a comment that it is a lower bound and in practice deprecations may take longer. Also, nothing stops us from lengthening the period later. WDYT?

Reply:
LGTM